### PR TITLE
STAC-12034: Add support for Kubernetes Secrets

### PR DESCRIPTION
**Pull request by hierynomus**
### Step 1: Link to Jira issue
STAC-12034

### Step 2: Description of changes
Add support for Kubernetes secrets to the topology

### Step 3: Did you add / update tests for your changes in the right area?
- [X] Unit/Component test
- [ ] Integration test
- [X] E2E/Molecule test

### Step 4: I'm confident that everything is properly tested:
I got a PO / QA Approval by:
- [ ] Name

### Step 5: Did you add release notes describing the changes you made?
- [ ] Yes

### Step 6: Can we ship this feature to production?
- [ ] Yes, I'm proud of my work. Ship it! :ship:

### CODE REVIEW

File: pkg/collector/corechecks/cluster/topologycollectors/secret_collector.go [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+			"creationTimestamp": secret.CreationTimestamp,
+			"tags":              tags,
+			"uid":               secret.UID,
+			"identifiers":       []string{secretExternalID},
+		},
+	}
+
+	component.Data.PutNonEmpty("generateName", secret.GenerateName)
+	component.Data.PutNonEmpty("kind", secret.Kind)
+	component.Data.PutNonEmpty("data", secret.Data)
```
Comment (rb3ckers) [inline pkg/collector/corechecks/cluster/topologycollectors/secret_collector.go:65]: Maybe for secrets we shouldn't include the data itself? It's only base64 encoded and probably contains passwords or other sensitive data

**Review by rb3ckers — CHANGES_REQUESTED** [review]

File: pkg/collector/corechecks/cluster/topologycollectors/secret_collector.go (hunk 1 above)
Comment (hierynomus) [reply pkg/collector/corechecks/cluster/topologycollectors/secret_collector.go:65]: A hash maybe for comparison purposes?

Comment (hierynomus) [reply pkg/collector/corechecks/cluster/topologycollectors/secret_collector.go:65]: i.e. we do want to do version/change detection I think

Comment (hierynomus) [reply pkg/collector/corechecks/cluster/topologycollectors/secret_collector.go]: Fixed this by hashing it before sending to Sts

**Review by rb3ckers — APPROVED** [review]
LGTM

**Review by zandrewitte — APPROVED** [review]

**Pull request merged**